progs/tests/string_conversion.sy

```diff
@@ -0,0 +1,67 @@
+A :: blob {
+    a: int,
+    b: float,
+}
+
+B :: blob {
+    self: B,
+    _name: void,
+}
+
+start :: fn {
+    // Basic types
+    as_str(-1) <=> "-1"
+    as_str(2.0) <=> "2.0"
+    as_str(2.00000000000182) <=> "2.00000000000182"
+    as_str("3.0") <=> "3.0"
+    as_str(true) <=> "true"
+    as_str(false) <=> "false"
+
+    // List
+    a := []
+    a -> push' a
+    a -> push' 1
+    as_str(a) <=> "[[...], 1]"
+
+    // Tuple
+    print ()
+    print (1,)
+    print (1, 2)
```

Comment:
Prints aren't really checkable - so these only add coverage of lines we don't check.